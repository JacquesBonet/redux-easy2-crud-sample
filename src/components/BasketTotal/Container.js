import BasketTotal from "./Component";
import {connect} from "react-redux";
import {getTotal, getProducts} from "../../services/products";

const mapStateToProps = (state) => ({
  total: getTotal(getProducts(state))
});

export default connect(mapStateToProps)(BasketTotal);
