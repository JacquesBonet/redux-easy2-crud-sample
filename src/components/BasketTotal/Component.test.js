import React from "react";
import {shallow} from "enzyme";
import BasketTotal from "./Component";

describe('BasketTotalComponent', function () {
  describe('render', function () {
    let component;
    const renderShallow = (props) => (shallow(<BasketTotal {...props} />));

    [[5.00, '$5.00'], [4.1, '$4.10']].forEach(function ([total, formatted]) {
      describe(`total is ${total}`, function () {
        beforeEach(function () {
          component = renderShallow({total});
        });
        it('should contain the correct total', function () {
          expect(component.html()).toMatchSnapshot();
        });
      });
    });
  });
});
