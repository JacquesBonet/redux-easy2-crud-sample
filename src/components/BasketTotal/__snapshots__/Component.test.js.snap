exports[`BasketTotalComponent render total is 4.1 should contain the correct total 1`] = `"<div class=\"app__basket-total\">$4.10</div>"`;

exports[`BasketTotalComponent render total is 5 should contain the correct total 1`] = `"<div class=\"app__basket-total\">$5.00</div>"`;
